lib/linter.js

```diff
@@ -308,6 +308,9 @@ class Linter {
     let hasBOM = options.source.charCodeAt(0) === 0xfeff;
     let source = hasBOM ? options.source.slice(1) : options.source;
 
+    // Strips doctype from the source if any
+    source = source.replace(/<!doctype[^>]+>\n?/gi, '');
```

Comment:
I don't believe we need the 'g' flag on this regex, as you're just replacing a single occurrence. You only need the 'g' flag if you're planning on looping/replacing multiple occurrences.

```js
let pattern = /a/g;
console.log(pattern.test('a')); // true
console.log(pattern.test('a')); // false
```